tests/unit/integrations/front-end/crawl-cleanup-searches-test.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+namespace Yoast\WP\SEO\Tests\Unit\Integrations\Front_End;
+
+use Brain\Monkey;
+use Mockery;
+use Yoast\WP\SEO\Conditionals\Front_End_Conditional;
+use Yoast\WP\SEO\Helpers\Options_Helper;
+use Yoast\WP\SEO\Integrations\Front_End\Crawl_Cleanup_Searches;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+use Yoast\WP\SEO\Helpers\Redirect_Helper;
+use WP_Query;
+
+/**
+ * Class Crawl_Cleanup_Searches_Test.
+ *
+ * @coversDefaultClass \Yoast\WP\SEO\Integrations\Front_End\Crawl_Cleanup_Searches
+ *
+ * @group integrations
+ * @group front-end
+ */
+class Crawl_Cleanup_Searches_Test extends TestCase {
+
+	/**
+	 * Robots helper.
+	 *
+	 * @var Mockery\MockInterface|Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * Redirect helper.
+	 *
+	 * @var Mockery\MockInterface|Redirect_Helper
+	 */
+	private $redirect_helper;
+
+	/**
+	 * The test instance.
+	 *
+	 * @var Crawl_Cleanup_Searches|Mockery\MockInterface
+	 */
+	private $instance;
+
+	/**
+	 * Sets an instance for test purposes.
+	 */
+	protected function set_up() {
+		parent::set_up();
+		$this->stubEscapeFunctions();
+
+		$this->options_helper  = Mockery::mock( Options_Helper::class );
+		$this->redirect_helper = Mockery::mock( Redirect_Helper::class );
+		$this->instance        = new Crawl_Cleanup_Searches( $this->options_helper, $this->redirect_helper );
+	}
+
+	/**
+	 * Tests if the expected conditionals are in place.
+	 *
+	 * @covers ::get_conditionals
+	 */
+	public function test_get_conditionals() {
+		$this->assertEquals(
+			[ Front_End_Conditional::class ],
+			Crawl_Cleanup_Searches::get_conditionals()
+		);
+	}
+
+	/**
+	 * Tests if the expected hooks are registered.
+	 *
+	 * @covers ::register_hooks
+	 */
+	public function test_register_hooks() {
+		$this->options_helper
+			->expects( 'get' )
+			->once()
+			->with( 'search_cleanup' )
+			->andReturn( true );
+
+		$this->options_helper
+			->expects( 'get' )
+			->once()
+			->with( 'redirect_search_pretty_urls' )
+			->andReturn( true );
+
+		Monkey\Functions\expect( 'get_option' )
+			->with( 'permalink_structure' )
+			->andReturn( '/%year%/%monthnum%/%postname%/' );
+
+		$this->instance->register_hooks();
+
+		$this->assertNotFalse( Monkey\Filters\has( 'pre_get_posts', [ $this->instance, 'validate_search' ] ) );
+		$this->assertNotFalse(
+			Monkey\Actions\has(
+				'template_redirect',
+				[
+					$this->instance,
+					'maybe_redirect_searches',
+				]
+			)
+		);
+	}
+
+	/**
+	 * Tests that nothing happens on a non search page.
+	 *
+	 * @covers       Crawl_Cleanup_Searches::validate_search
```

Comment:
This is not found due to missing namespace, but let's fall back to the `@coversDefaultClass`.

```suggestion
	 * @covers ::validate_search
```